editor/header/index.js

```diff
@@ -1,12 +1,23 @@
 /**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+
+/**
  * Internal dependencies
  */
 import './style.scss';
+import MultiSelectHeader from './multi-select';
 import ModeSwitcher from './mode-switcher';
 import SavedState from './saved-state';
 import Tools from './tools';
+import { getSelectedBlocks } from '../selectors';
+
+function Header( { selectedBlocks } ) {
+	if ( selectedBlocks.length ) {
+		return <MultiSelectHeader selectedBlocks={ selectedBlocks } />;
```

Comment:
Depending to which extreme we want to take component self-sufficiency, `<MultiSelectHeader />` is equally capable of retrieving the currently selected blocks. We could even adopt a pattern where we always render `<MultiSelectHeader />`, but it returns `null` if there's not a multiple selection.

Not really suggesting any change here, just food for thought.